pkg/monitor/datapath_drop.go

```diff
@@ -34,38 +37,53 @@ type DropNotify struct {
 	Hash     uint32
 	OrigLen  uint32
 	CapLen   uint32
-	SrcLabel uint32
-	DstLabel uint32
+	SrcLabel identity.NumericIdentity
+	DstLabel identity.NumericIdentity
 	DstID    uint32
 	Unused   uint32
 	// data
 }
 
+// dumpIdentity dumps the source and destination identities in numeric or
+// human-readable format.
+func (n *DropNotify) dumpIdentity(buf *bufio.Writer, numeric bool) {
+	if numeric {
+		fmt.Fprintf(buf, "identity %d->%d", n.SrcLabel, n.DstLabel)
+	} else {
+		fmt.Fprintf(buf, "identity %s->%s", n.SrcLabel, n.DstLabel)
+	}
+}
+
 // DumpInfo prints a summary of the drop messages.
-func (n *DropNotify) DumpInfo(data []byte) {
-	fmt.Printf("xx drop (%s) flow %#x to endpoint %d, identity %d->%d: %s\n",
-		api.DropReason(n.SubType), n.Hash, n.DstID, n.SrcLabel, n.DstLabel,
-		GetConnectionSummary(data[DropNotifyLen:]))
+func (n *DropNotify) DumpInfo(data []byte, numeric bool) {
+	buf := bufio.NewWriter(os.Stdout)
+	fmt.Fprintf(buf, "xx drop (%s) flow %#x to endpoint %d, ",
+		api.DropReason(n.SubType), n.Hash, n.DstID)
+	n.dumpIdentity(buf, numeric)
+	fmt.Fprintf(buf, ": %s\n", GetConnectionSummary(data[DropNotifyLen:]))
+	buf.Flush()
 }
 
 // DumpVerbose prints the drop notification in human readable form
-func (n *DropNotify) DumpVerbose(dissect bool, data []byte, prefix string) {
-	fmt.Printf("%s MARK %#x FROM %d DROP: %d bytes, reason %s",
+func (n *DropNotify) DumpVerbose(dissect bool, data []byte, prefix string, numeric bool) {
+	buf := bufio.NewWriter(os.Stdout)
```

Comment:
Same comment as previous: pass in an `io.Writer` instead of hardwiring `os.Stdout`.